src/java/com/zimbra/qa/unittest/TestSSDBEphemeralStore.java

```diff
@@ -35,298 +39,279 @@
         } else {
             SSDBStoreConfigured = false;
         }
     }
 
     @Override
     public void tearDown() throws Exception {
-        if(SSDBStoreConfigured) { //switch to assumptions when we upgrade test suite to Junit 4
-            String ssdbUrl = Provisioning.getInstance().getConfig().getEphemeralBackendURL();
-            String toks[] = ssdbUrl.split(":");
-            SSDBEphemeralStore.getFactory().shutdown();
-            try {
-                try(JedisPool pool = new JedisPool(toks[1], Integer.parseInt(toks[2]))) {
-                    try (Jedis jedis = pool.getResource()) {
-                        jedis.flushDB();
-                    }
+        if (SSDBStoreConfigured) {
+            for (Pair<EphemeralInput, EphemeralLocation> pair: toDelete) {
+                EphemeralInput input = pair.getFirst();
+                EphemeralLocation location = pair.getSecond();
+                try {
+                    store.delete(input.getEphemeralKey(), (String) input.getValue(), location);
+                } catch (ServiceException e) {
+                    // ignore failed deletions
                 }
-            } catch (ClassCastException e) {
-                //ignore
             }
+            toDelete.clear();
+            SSDBEphemeralStore.getFactory().shutdown();
         }
     }
 
+    private void addDeletionEntry(EphemeralInput input, EphemeralLocation location) {
+        toDelete.add(new Pair<EphemeralInput, EphemeralLocation>(input, location));
+    }
+
+    private void doSet(EphemeralInput input, EphemeralLocation location) throws ServiceException {
+        store.set(input, location);
+        addDeletionEntry(input, location);
+    }
+
+    private void doUpdate(EphemeralInput input, EphemeralLocation location) throws ServiceException {
+        store.update(input, location);
+        addDeletionEntry(input, location);
+    }
+
     @Test
     public void testSetGetLastLogin() throws Exception {
-        if(SSDBStoreConfigured) { //switch to assumptions when we upgrade test suite to Junit 4
-            String firstLogin = "20160912212057.178Z";
-            EphemeralStore store = SSDBEphemeralStore.getFactory().getStore();
-            assertTrue(store instanceof SSDBEphemeralStore);
-            EphemeralLocation accountIDLocation = new EphemeralLocation() {
-                @Override
-                public String[] getLocation() { return new String[] { "account", ACCOUNT_ID }; }
-            };
-            EphemeralKey eKey = new EphemeralKey(Provisioning.A_zimbraLastLogonTimestamp);
-            EphemeralInput attr = new EphemeralInput(eKey, firstLogin);
-            store.set(attr, accountIDLocation);
-
-            EphemeralResult retAttr = store.get(eKey, accountIDLocation);
-            assertNotNull(retAttr);
-            assertEquals("Found incorrect last logon timestamp value",firstLogin, retAttr.getValue());
-        }
+        Assume.assumeTrue(SSDBStoreConfigured);
```

Comment:
Assume works in the setup method, so you can eliminate un-needed and duplicate code by moving it there.
Also, I was puzzled and annoyed that Assume didn't have a way to specify a useful description of what assumption had failed, so I found a hackaround for that - See `TestUtil.assumeTrue(String missive, Boolean testVal)`.  Please use that as again, you will be able to just look at the summary report and immediately see why this test was skipped instead of getting some obscure "false wasn't true" message without any mention of what was meant to be true.  Thinking of trying to get more AssumeTrue method signatures into JUnit at some point.